### DISCUSSION TITLE
Delayed starts from 2 points with different delay times

### DISCUSSION THREAD

**Discussion by martoconnh** [Parcels-code/Parcels#1457] [Q&A]
Hi! I have 2 source points (2 river estuaries) from where my particles are "dropped" in my simulation. I was wondering if I can make new particles appear in each source, but with different delay times.

Basically, I have a different particle release ratio each day for each of my source points. For instance, on day 1, point 1 drops particles every 60 seconds while point 2 drops them every 30 seconds. Then on day 2, point 1 drops them every 125 seconds and point 2 every 80 seconds, and so on for a few weeks (I have .dat file containing the particle release ratio for each day).

I have looked it up here:
https://docs.oceanparcels.org/en/latest/examples/tutorial_delaystart.html
but have not found anything about different delay times.

For now I have been releasing new particles from each point using:
```
repeatdt = delta(hours=1)
pset = ParticleSet(fieldset=fieldset, pclass=Particula, lon=lons, lat=lats, repeatdt=repeatdt)
```

Is there any way I can assign a different delay time to each one of my source points?

**Comment by michaeldenes** [comment]
Hi @martoconnh, You can specify the times to release the particles
e.g. `pset = ParticleSet.from_list(fieldset=fieldset, pclass=Particula, lon=lons, lat=lats, time=times)`
where `times` is an array of particle start times for your particular particles.
Let me know how that goes!

**Reply by erikvansebille** [reply]
Indeed, see this section of the tutorial: https://docs.oceanparcels.org/en/latest/examples/tutorial_delaystart.html#Assigning-each-particle-its-own-time. 

So don't use `repeatdt` at all.

Note that you would then need to define N+M different particles, if you want N particles to start from point 1 and M particles from point 2.

**Comment by martoconnh** [comment]
Hi! Just letting you know everything worked out! Thank you so much!!

However, I'm given the following warning several times every time I run my program (not new, it always happened before making these changes). 

`<__array_function__ internals>:200: RuntimeWarning: invalid value encountered in cast`

Appart from that, the simulation works just fine and my results are coherent with what I was expecting, so I did not worry a lot about it. I was just wondering what did this warning meant.

**Reply by erikvansebille** [reply]
Which Parcels version do you use? This casting warning was an issue with the old v2.4, which has been fixed in https://github.com/OceanParcels/parcels/pull/1400 and should now have disappeared in the latest Parcels v3.0.

**Reply by martoconnh** [reply]
I'm using Parcels v2.4.2. I have just updated it a few minutes ago using `conda update parcels` in my anaconda terminal. But it's still v2.4.2. Also, as I was updating it I got this message in my terminal:

```
Collecting package metadata (current_repodata.json): done
Solving environment: /
The environment is inconsistent, please check the package plan carefully
The following packages are causing the inconsistency:

  - defaults/win-64::anaconda-catalogs==0.2.0=py311haa95532_0
  - defaults/win-64::anaconda-cloud-auth==0.1.3=py311haa95532_0
  - defaults/win-64::anaconda-navigator==2.5.0=py311haa95532_0
  - defaults/win-64::black==23.3.0=py311haa95532_0
  - defaults/win-64::conda-build==3.26.1=py311haa95532_0
  - defaults/win-64::conda-index==0.3.0=py311haa95532_0
  - defaults/noarch::conda-verify==3.4.2=py_1
  - defaults/noarch::cookiecutter==1.7.3=pyhd3eb1b0_0
  - defaults/win-64::dask==2023.6.0=py311haa95532_0
  - defaults/win-64::dask-core==2023.6.0=py311haa95532_0
  - defaults/win-64::datashader==0.15.2=py311haa95532_0
  - defaults/win-64::distributed==2023.6.0=py311haa95532_0
  - defaults/win-64::flask==2.2.2=py311haa95532_0
  - defaults/win-64::intake==0.6.8=py311haa95532_0
  - defaults/win-64::nltk==3.8.1=py311haa95532_0
  - conda-forge/noarch::parcels==2.4.2=pyh7428d3b_2
  - defaults/win-64::python-dotenv==0.21.0=py311haa95532_0
  - defaults/win-64::python-lsp-black==1.2.1=py311haa95532_0
  - defaults/win-64::scikit-image==0.20.0=py311h3513d60_0
  - defaults/win-64::spyder==5.4.3=py311haa95532_1
  - conda-forge/noarch::trajan==0.5.1=pyhd8ed1ab_0
  - defaults/win-64::_anaconda_depends==2023.09=py311_mkl_1
done
```
